daemon/daemon.h

```diff
@@ -5,13 +5,55 @@
 
 #pragma once
 
-struct tr_error;
+#include <string>
+#include <string_view>
 
-typedef struct dtr_callbacks
+#include <libtransmission/transmission.h>
+#include <libtransmission/variant.h>
+#include <libtransmission/error.h>
+#include <libtransmission/utils.h>
+#include <libtransmission/file.h>
+#include <libtransmission/log.h>
+
+using namespace std::literals;
+
+class tr_daemon
 {
-    int (*on_start)(void* arg, bool (*setupsigfn)(void*), bool foreground);
-    void (*on_stop)(void* arg);
-    void (*on_reconfigure)(void* arg);
-} dtr_callbacks;
+#ifdef HAVE_SYS_SIGNALFD_H
```

Comment:
IMO, it's better to have `public:` first and then define `private:` section explicitly after

Reply:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rl-order

Reply:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rl-order